lib/trino-hive-formats/src/main/java/io/trino/hive/formats/ion/IonDecoderFactory.java

```diff
@@ -166,6 +206,65 @@ private static BlockDecoder wrapDecoder(BlockDecoder decoder, Type trinoType, Io
         };
     }
 
+    /**
+     * Coerces an Ion value to its string representation.
+     *
+     * This method handles all IonTypes and converts them to a string format.
+     * For complex types (LIST, SEXP, STRUCT), it recursively processes their elements.
+     *
+     * @param reader The IonReader containing the value to be coerced.
+     * @param type The IonType of the value to be coerced.
+     * @return A string representation of the Ion value.
+     * @throws IllegalArgumentException if the IonType is not supported for text coercion.
+     * @throws IonException if there's an error reading from the IonReader.
+     */
+    private static String coerceToString(IonReader reader, IonType type)
```

Comment:
The legacy hive impl "just" uses `IonValue.toString()` (IIRC), that may be the simplest way to do this.

For this functionality I'd value backwards compatibility and simplicity above all else.

Reply:
I initially thought to use `IonValue` but in order to get an `IonValue` from the reader I need to use `IonSystem` and then pass the reader to get an iterator over `IonValue` but the reader would already have moved to next value at this point and wouldn't give us the current `IonValue` which is why I chose this manual approach.

Reply:
I see. I always get a little twisted up in this area, but I think another approach would be to create an `IonWriter` and then `write(ionReader)` to a buffer. I just don't want to go down a path of trying to replicate the printing that we have today. I also want to keep the door open to having a JSON downconversion flag.

Reply:
That worked perfectly fine. Changed the code to use writer instead of the manual approach.